src/answer/answer.service.ts

```diff
@@ -1,147 +1,259 @@
+        });
+        if (answer == undefined){
+            throw new AnswerNotFoundError(answerId);
+        }
+        const userDto = await this.usersService.getUserDtoById(userId);
+        return {
+            id: answer.id,
+            question_id: answer.question_Id,
+            content: answer.content,
+            author: userDto,
+            created_at: answer.createdAt.getTime(),
+            updated_at: answer.updatedAt.getTime(),
+            // agree_type: answer.agree_type,
+        };
+    }
 
-  async agreeAnswer(id: number, userId: number): Promise<AnswerDto> {
-    const answer = await this.answerRepository.findOneBy({ id });
+    async updateAnswer(
+      userId: number,
+      answerId: number,
+      content: string,
+    ): Promise<void> {
+        const answer = await this.answerRepository.findOne({ where: { id: answerId } });
 
-    if (!answer) {
-      throw new AnswerNotFoundError(id);
+        if (!answer) {
+          throw new AnswerNotFoundError(answerId);
+        }
+    
+        // Update the answer with the data from the DTO
+        answer.content = content;
+    
+        // Save the updated answer
+        await this.answerRepository.save(answer);
     }
 
-    if (answer.agrees && answer.agrees.includes(userId)) {
-      throw new AnswerAlreadyAgreeError(id);
+    async deleteAnswer(
+        userId: number,
+        questionId: number,
+        answerId: number,
+    ): Promise<void> {
+        const answer = await this.answerRepository.findOne({
+            where: {id: answerId},
+        });
+        if (!answer){
+            throw new AnswerNotFoundError(answerId);
+        }
+        await this.answerRepository.delete(answerId);
     }
 
-    if (!answer.agrees) {
-      answer.agrees = [userId];
-    } else {
-      if (answer.agree_type == 1) {
-        answer.agrees.push(userId);
-        answer.agree_count = (answer.agree_count || 0) + 1;
-      } else {
-        answer.disagrees.push(userId);
-        answer.disagree_count = (answer.disagree_count || 0) + 1;
-      }
-    }
-    return this.answerRepository.save(answer);
-  }
+    async agreeAnswer(id: number, userId: number, agree_type: number): Promise<AgreeAnswerDto> {
+        // 查找要点赞的答案
+        const answer = await this.answerRepository.findOne({ where: { id }});
 
-  async favoriteAnswer(id: number, userId: string): Promise<Answer> {
-    // 使用提供的 answerId 查询数据库中的特定答案实体
-    const answer = await this.answerRepository.findOne({ where: { id } });
+        if (!answer) {
+            throw new AnswerNotFoundError(id);
+        }
 
-    // 检查答案是否存在
-    if (!answer) {
-      throw new AnswerNotFoundError(id);
-    }
+        // 检查用户是否已经点过赞
+        if (answer.agrees && answer.agrees.includes(userId)) {
+            throw new AnswerAlreadyAgreeError(id);
+        }
 
-    if (answer.is_favorite && answer.favoritedBy.includes(userId)) {
-      throw new AnswerAlreadyFavoriteError(id);
+        // 更新点赞信息
```

Comment:
Do not write comments unless truly necessary. See [Don't Write Comments](https://www.youtube.com/watch?v=Bf7vDBBOBUA)
More importantly, Chinese characters forbidden.